jenkins_pipelines/environments/common/pipeline-build-validation-cleanup.groovy

```diff
@@ -0,0 +1,249 @@
+        GString resultdir = "${WORKSPACE}/results"
+        GString resultdirbuild = "${resultdir}/${BUILD_NUMBER}"
+        GString exports = "export BUILD_NUMBER=${BUILD_NUMBER}; export BUILD_VALIDATION=true; "
+        String container_repository = params.container_repository ?: null
+        String product_version = null
+        String controllerHostname = null
+        String serverHostname = null
+        GString targetedTfFile = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/main.tf"
+        GString targetedTfStateFile = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/terraform.tfstate"
+        GString targetedTerraformDirPath = "/home/jenkins/workspace/${params.targeted_project}/results/sumaform/"
+        GString localSumaformDirPath = "${resultdir}/sumaform/"
+        GString localTfStateFile = "${localSumaformDirPath}/terraform.tfstate"
+        GString logFile = "${resultdirbuild}/sumaform.log"
+
+        // Construct the --tf-resources-to-delete argument dynamically
+        ArrayList defaultResourcesToDelete = []
+        if (params.clean_proxy) {
+            defaultResourcesToDelete.add('proxy')
+        }
+        if (params.clean_monitoring_server) {
+            defaultResourcesToDelete.add('monitoring-server')
+        }
+        if (params.clean_retail) {
+            defaultResourcesToDelete.add('retail')
+        }
+
+        String defaultResourcesToDeleteArgs = defaultResourcesToDelete.isEmpty() ? '' : "--default-resources-to-delete ${defaultResourcesToDelete.join(' ')}"
+
+        GString commonParams = "--outputdir ${resultdir} --tf ${targetedTfFile} --gitfolder ${resultdir}/sumaform"
+
+        if (params.terraform_parallelism) {
+            commonParams = "${commonParams} --parallelism ${params.terraform_parallelism}"
+        }
+
+        try {
+            stage('Clone terracumber, susemanager-ci and sumaform') {
+
+                // Prevent rebuild option
+                if (currentBuild.getBuildCauses().toString().contains("RebuildCause")) {
+                    error "Rebuild is blocked for this job."
+                }
+
+                // Create a directory for  to place the directory with the build results (if it does not exist)
+                sh "mkdir -p ${resultdir}"
+                git url: params.terracumber_gitrepo, branch: params.terracumber_ref
+                dir("susemanager-ci") {
+                    checkout scm
+                }
+
+                // Clone sumaform
+                sh "set +x; source /home/jenkins/.credentials set -x; ./terracumber-cli ${commonParams} --gitrepo ${params.sumaform_gitrepo} --gitref ${params.sumaform_ref} --runstep gitsync"
+
+                // Restore Terraform states from artifacts
+                if (params.use_previous_terraform_state) {
+                    copyArtifacts projectName: currentBuild.projectName, selector: specific("${currentBuild.previousBuild.number}")
+                }
```

Comment:
Actually... using the previous terraform state is mandatory for this pipeline right?
Otherwise we can't get the metadata to do any cleanup 🤔

Reply:
This parameter is not needed and possible. Removed.